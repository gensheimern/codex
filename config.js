module.exports = {
	DB_HOST: process.env.DB_HOST || '159.65.115.221',
	DB_USER: process.env.DB_USER || 'root',
	DB_PASS: process.env.DB_PASS || 'teamcodex',
	DB_NAME: process.env.DB_NAME || 'lunch_planner',
	JWT_SECRET: process.env.JWT_SECRET || 'secret',
	MAIL_HOST: process.env.MAIL_HOST || '159.65.115.221',
	MAIL_USER: process.env.MAIL_USER || 'support@codex-team.de',
	MAIL_PASS: process.env.MAIL_PASS || 'teamcodex123',
};
