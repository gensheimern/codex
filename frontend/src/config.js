const config = {
	debug: process.env.REACT_APP_DEBUG === 'TRUE',
	apiPath: process.env.REACT_APP_API_PATH,
	wsPath: process.env.REACT_APP_WS_PATH,
	basePath: process.env.REACT_APP_BASE_PATH,
}

export default config;