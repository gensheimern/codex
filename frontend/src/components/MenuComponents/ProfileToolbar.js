import React from "react";
import Toolbar, { ToolbarGroup } from 'material-ui/Toolbar';
import Avatar from 'material-ui/Avatar';
import Notification from 'material-ui/svg-icons/social/notifications';
import { Link, withRouter } from 'react-router-dom';
import notificationChecker from '../notification/notificationChecker';
import Badge from 'material-ui/Badge';
import config from '../../config';

class ProfileToolbar extends React.Component {

	constructor(props) {
		super(props);
		this.state = {
			newNotifications: 0,
			unseenNotifications: 0,
			unsubscribed: false,
		};
	}

	componentDidMount() {
		notificationChecker.enable('notification', () => {
			if (!this.state.unsubscribed) {
				this.setState((prevState, props) => ({
					newNotifications: prevState.newNotifications + 1,
				}));
			}
		});

		this.loadUnseenNotifications();
	}

	loadUnseenNotifications() {
		fetch(config.apiPath + "/user/me/notifications/unseen", {
            method: 'GET',
            headers: {
				'Content-Type': 'application/json',
				'X-Access-Token': localStorage.getItem('apiToken'),
            }
		})
		.then((res) => {
            if(!res.ok) {
                throw new Error("Response not ok.");
            } else if(res.status !== 200) {
                throw new Error("An error occured.");
            }
            return res;
		})
		.then(res => res.json())
		.then((res) => {
            this.setState({
				unseenNotifications: res.unseenNotifications,
			});
        }).catch((error) => {
			this.setState({
				unseenNotifications: 0,
			});
        });
	}

	componentWillUnmount() {
		this.setState({
			unsubscribed: true,
		});
		notificationChecker.disable('notification');
	}

	readNotifications = () => {
		this.setState({
			newNotifications: 0,
			unseenNotifications: 0,
		});
	}

	render(){
		const notificationIcon = this.state.newNotifications + this.state.unseenNotifications > 0
			? 	(<Badge
					badgeContent={this.state.newNotifications + this.state.unseenNotifications}
					badgeStyle={{
						backgroundColor: 'red',
						color: 'white',
						top: 12,
						right: 12,
					}}
				>
					<Notification onClick={() => {
						this.readNotifications();
						this.props.history.push('/notifications');
					}}
					style={{
						cursor: 'pointer',
					}}/>
				</Badge>)
			:	(<Notification onClick={() => {
					this.readNotifications();
					this.props.history.push('/notifications');
				}}/>);

		return (
			<React.Fragment>
				<Toolbar style={{
					backgroundColor: 'white',
				}}>
					<ToolbarGroup firstChild={true} style={{
						marginLeft: 0,
					}}>
					{notificationIcon}
					</ToolbarGroup>

					<ToolbarGroup>
						<Link to="/profile">
							<Avatar>MM</Avatar>
						</Link>
					</ToolbarGroup>
				</Toolbar>
			</React.Fragment>
		);
	}

}

export default withRouter(ProfileToolbar);
