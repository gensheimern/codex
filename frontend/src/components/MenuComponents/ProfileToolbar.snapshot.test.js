/**
 * @author:Stella Neser
 *  Komponente ist ein Snapshottest
 */
import React from 'react';
import renderer from'react-test-renderer';
import ProfileToolbar from './ProfileToolbar';
import config from '../../config';
import {MemoryRouter} from 'react-router-dom';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
const fetchMock = require('fetch-mock');

beforeEach(() => {
    window.localStorage = {getItem: () => ""};
    fetchMock.get(config.apiPath +'/team', 
    [ { id: 6,
		name: "DÃ¶ner",
		manager: {
			id: 6,
		    firstName: "Max",
		    name: "Mustermann",
		    email: "max.mustermann@web.de",
		    image: "",
		    me: true,
		},
    }]);

    fetchMock.get(config.apiPath +'/user/me/',
    {
        id: 6,
        firstName: 'Max',
        name: 'Mustermann',
        email: 'max.mustermann@gmail.com',
        image: '',
        me: true,
    });

    fetchMock.get(config.apiPath +'/user/me/notifications/unseen', {
        unseenNotifications: 4,
		
    });

    fetchMock.get(config.apiPath +'/organization/', {
        id: 8,
		name: "Test",
		description: "Dies ist ein Test",
		admin: {
            id: 6,
            firstName: 'Max',
            name: 'Mustermann',
            email: 'max.mustermann@gmail.com',
            image: '',
            me: true,
		},
		
    });
    fetchMock.get(config.apiPath +'/user/me/organizations', {
        id: 8,
		name: "Test",
		description: "Dies ist ein Test",
		admin: {
            id: 6,
            firstName: 'Max',
            name: 'Mustermann',
            email: 'max.mustermann@gmail.com',
            image: '',
            me: true,
		},
		
    });
    
});

afterEach(() => {
    fetchMock.restore();
});

describe('ProfileToolbar Snapshot', () => {
    test('renders', () =>{
        const component = renderer.create(
            <MemoryRouter>
                <MuiThemeProvider>
                    <ProfileToolbar/>
                </MuiThemeProvider>
            </MemoryRouter>
        );
        let tree= component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});