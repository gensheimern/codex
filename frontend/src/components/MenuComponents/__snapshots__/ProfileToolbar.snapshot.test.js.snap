// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileToolbar Snapshot renders 1`] = `
<div
  className={undefined}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "white",
      "boxSizing": "border-box",
      "display": "flex",
      "height": 56,
      "justifyContent": "space-between",
      "muiPrepared": true,
      "padding": "0px 24px",
    }
  }
>
  <div
    className={undefined}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
        "marginLeft": 0,
        "marginRight": undefined,
        "muiPrepared": true,
        "position": "relative",
      }
    }
  >
    <svg
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "inline-block",
          "fill": "currentColor",
          "height": 24,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "width": 24,
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
      />
    </svg>
  </div>
  <div
    className={undefined}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
        "marginLeft": undefined,
        "marginRight": undefined,
        "muiPrepared": true,
        "position": "relative",
      }
    }
  >
    <div
      className={undefined}
      onClick={[Function]}
      size={40}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(188, 188, 188)",
          "borderRadius": "50%",
          "color": "#ffffff",
          "cursor": "pointer",
          "display": "inline-flex",
          "fontSize": 20,
          "height": 40,
          "justifyContent": "center",
          "muiPrepared": true,
          "userSelect": "none",
          "width": 40,
        }
      }
    />
    <div
      style={
        Object {
          "display": "none",
        }
      }
    />
  </div>
</div>
`;
